src/libcrun/linux.c

```diff
@@ -516,12 +518,39 @@ parse_idmapped_mount_option (runtime_spec_schema_config_schema *def, bool is_uid
   return 0;
 }
 
+static char *
+format_mount_mappings (runtime_spec_schema_defs_id_mapping **mappings, size_t mappings_len, size_t *written)
+{
+  char buffer[64];
+  char *ret;
+  size_t s;
+
+  *written = 0;
+
+  ret = xmalloc (sizeof (buffer) * mappings_len + 1);
+  for (s = 0; s < mappings_len; s++)
+    {
+      size_t len;
+
+      /* Invert host and container id when writing the mappings.  */
+      len = sprintf (buffer, MAPPING_FMT_SIZE,
```

Comment:
nit: maybe `snprintf(buffer, sizeof(buffer), ...` although it's clear now that it can't be overflown, the code might change later (hypothetically, say we'll switch to uint64)